lenskit-api/src/main/java/org/lenskit/api/ItemBasedItemRecommender.java

```diff
@@ -29,62 +26,68 @@
 
 
 /**
- * The interface for recommendation based on the items only. The difference from {@link
- * ItemRecommender} is that the input is only the item or list of items instead of user specific
- * information. This interface can be used to provide a Find Similar Items / People Also Liked
- * feature.
+ * Recommends items that go with a set of reference items. This interface is distinguished from
+ * {@link ItemRecommender} in that it uses a set of reference items instead
+ * of a user as the basis for computing scores. This component can be used to provide a Find Similar Items
+ * or People Also Liked feature.
  *
  * @see ItemRecommender
  * @author <a href="http://www.grouplens.org">GroupLens Research</a>
  * @compat Public
  * @since 0.10
  */
-public interface GlobalItemRecommender {
+public interface ItemBasedItemRecommender {
     /**
      * Recommend all possible items for a basket of items using the default exclude set.
      *
-     * @param items The items value.
+     * @param basket The reference basket.
      * @return The recommended items.
-     * @see #globalRecommend(Set, int, Set, Set)
+     * @see #recommendRelatedItems(Set, int, Set, Set)
      */
-    List<ScoredId> globalRecommend(Set<Long> items);
+    List<Long> recommendRelatedItems(Set<Long> basket);
 
     /**
      * Recommend up to <var>n</var> items for a basket of items using the default exclude set.
```

Comment:
If we implement a candidate identifier component, we can then document things as follows:

- The candidate/exclude-set-free methods call the complete method with candidate and exclude sets of `null`.
- The default recommender implementations pass their candidate and exclude sets, as-is, to a `findCandidates(long, LongSet, LongSet)` on `UserBasedCandidateIdentifier` or `findCandidates(LongSet, LongSet, LongSet)` on `ItemBasedCandidateIdentifier` method.
- The default `CandidateIdentifier` implements the following logic:

    - If the candidate set is `null`, use the full set of items from the `ItemDAO`
    - If the exclude set is `null`, use the reference set (for by-item) or the set of rated items (for rating-based by-user) or the set of touched items (for event-based by-user)
    - Return `candidates - exclude`